### ISSUE TITLE
[DEFECT] Unexpected behavior when assigning more than 5 employees to work simultaneously in the same workflow

### ISSUE THREAD

**Issue by louislab** [COMP3122-2324-Project/Simpsyber#7]
### Description
As the game didn’t mention anything about the limit on how many employees can be assigned to work simultaneously in the same workflow. To utilize the power of all the employees during the starting round, I managed to let them (6 people) work at the same time to design the apps. However, the game disallowed me to do that, which is unexpected. 

Follow the video below to trigger the issue:

https://github.com/COMP3122-2324-Project/Simpsyber/assets/80302893/ee40bab6-8b00-44d1-b665-483abce88be1

### Findings
I found that for each column in the Kanban board, there is a space limitation. Player can only assign up to 5 post-it memos in each column. But there exists 6 employees, that mean, for the starting round, I can only assign 5 employees to work, and the one who had been left behind can just sit there and wait, which is not ideal and align with the game settings.

### Environment
This issue can be reproduced in the following browsers:
Chrome Version 121.0.6167.184 (Official Build) (x86_64)
Safari Version 17.2.1 (19617.1.17.11.12)

Operating System:
macOS Sonoma Version 14.2.1

**Comment by chunyinng078** [comment]
I agree with your point that the game logic could be clearer about how many numbers of tasks could be at the same category(TODO/DESIGN, etc...).
Therefore I have two suggestions about how this defect could be solved.

Suggestion 1:
When users/players first enter the page, the information of "numbers of tasks in the same category could be done in the same time" should also be included in the hints like the other hints. An example is as follows:
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/95467201/8dd50dc6-4c53-4809-af26-84121932ddec)


Suggestion 2:
Add an indicator on the top of each category (or column)). So, users/players could easily know how many tasks they can have in the category (or column). An example is as follows:
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/95467201/fd5f022c-c788-44c9-8a54-daacab503599)

**Comment by 21027485D** [comment]
I second the proposal given by @chunyinng078. I also think that only able to assign 5 members in each step is intentional. It's probably designed to teach students about workflow congestions. I prefer suggestion 2 proposed by @chunyinng078 since it's clearer and intutive.

**Comment by gg11365** [comment]
I think this is indeed a bug, and not intentional as @21027485D  says. An interesting event is when one of your workflows is already fully allocated and the game suddenly has multiple "defective" events, you may not be able to continue the game. I consider this a serious defect.

https://github.com/COMP3122-2324-Project/Simpsyber/assets/85055782/969c9125-d8cc-40e7-84c8-9fdf4fe0b5f1

**Comment by YIPTakWa** [comment]
Hey there!

Thanks a lot for bringing up this issue! I totally agree with the point you made about the lack of a prompt in the game regarding the restriction of processing more than 5 tasks in the same column. It does seem illogical without any prior indication.

In my opinion, the solution could be to remove this restriction instead of adding a reminder to the player before the game starts.

In real-life scenarios, if a company has six tasks and six available employees, it's possible to assign different tasks to different employees simultaneously. This would increase the freedom in the game, allowing players to manage their tasks more flexibly. Therefore, the task column of this game must be written in the scroll view. Because when there are more than five column tasks, there is no place on the page to display the sixth task.
<img width="967" alt="螢幕截圖 2024-03-05 上午12 06 49" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/156769265/36628f66-660d-41e6-84b5-4cddb26e4eec">


Of course, we need to carefully consider the impact of such a change on the overall gameplay experience. Removing this limitation might add complexity to the game, requiring more strategic thinking and planning. We need to strike a balance and ensure that this change doesn't disrupt the game's balance and other aspects of the experience.
